db_test.go

```diff
@@ -935,6 +935,55 @@ func TestDB_Retention(t *testing.T) {
 	testutil.Equals(t, int64(100), db.blocks[0].meta.MaxTime) // To verify its the right block.
 }
 
+func TestSizeBasedRetention(t *testing.T) {
+	db, close := openTestDB(t, &Options{
+		BlockRanges: []int64{100},
+	})
+	defer close()
+
+	// Add some empty blocks.
+	blocks := []*BlockMeta{
+		{ULID: ulid.MustNew(100, nil), MinTime: 100, MaxTime: 200},
+		{ULID: ulid.MustNew(200, nil), MinTime: 200, MaxTime: 300},
+		{ULID: ulid.MustNew(300, nil), MinTime: 300, MaxTime: 400},
+		{ULID: ulid.MustNew(400, nil), MinTime: 400, MaxTime: 500},
+		{ULID: ulid.MustNew(500, nil), MinTime: 500, MaxTime: 600},
+	}
+	for _, m := range blocks {
+		createEmptyBlock(t, filepath.Join(db.Dir(), m.ULID.String()), m)
+	}
+
+	// Figure out how large an empty block is.
+	emptyBlockDir := filepath.Join(db.Dir(), blocks[0].ULID.String())
+	emptyBlock, err := OpenBlock(emptyBlockDir, db.chunkPool)
+	testutil.Ok(t, err)
+	db.opts.MaxBytes = 4 * emptyBlock.Size()
+
+	// Reload and ensure that actual size is less than limit.
+	testutil.Ok(t, db.reload())
+
+	var size int64
+	for _, b := range db.blocks {
+		size += b.Size()
+	}
+	testutil.Assert(t, size <= db.opts.MaxBytes, "actual size (%v) is expected to be less than or equal to limit (%v)", size, db.opts.MaxBytes)
+	testutil.Equals(t, 4, len(db.blocks))
```

Comment:
another comment here as well, why are we checking this?

Reply:
I can't tell, are you actually asking me why I'm checking this? Or just that the comment should answer that question?

Reply:
yeah just a comment to specify why are we checking this.
 It is clear for me and you, but for someone not familiar with the test will wonder.